### PR TITLE
React/Kanban Board

### PR DESCRIPTION
**Pull request by Awsaf780**
Fixes #

- [x] Does not depend on other PRs
- [ ] Depends on other PR being merged first: #

## What kind of change does this PR introduce (check at least one)

- [ ] Bugfix
- [ ] Improvement
- [x] New feature
- [ ] Refactor
- [x] Infrastructure
- [ ] Other (please specify)

## How would you describe the changes in a changelog (1-2 lines)?
- Add tasks to Kanban Board
- Drag and Drop tasks between boards
- Toggle Theme

## What was done
- Use Context API and useReducer to manage global state
- Toggle Theme change between Light and Dark on click
- Maintain count of items in each board
- Use Emotion styled components
- Use React-dnd to implement drag and drop functionality
- Organize file structure will small components

## How to test

- [ ] Needs regression testing (add QA needed label)

## Deploy risk & follow up

Please assess risk & follow up on deployment of this PR. Elaborate your choice if selecting anything other than 'No risk'.

- [x] No risk
- [ ] Low
- [ ] High

## Reminder list

- Determine whether REST API Documentation should be updated
- Make an assessment on whether unit tests should be updated and/or added
- Ensure follow up tasks are linked to this PR

QA - if regression testing was requested:

- [ ] QA has verified

### CODE REVIEW

File: react/kanban-board/src/App.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import { HTML5Backend } from 'react-dnd-html5-backend';
+import { AppDiv } from "./App-styled";
+import Home from "./components/home/Home";
+import KanbanState from "./contexts/State";
+
+function App() {
+
+  return (
+    <DndProvider backend={HTML5Backend}>
+        <AppDiv>
```
[the remark this answers is not in the document]
Comment (Awsaf780) [reply react/kanban-board/src/App.js]: Resolved in commit [04495fb](https://github.com/CraftsmenLtd/coe-batch-q322/pull/290/commits/04495fb29341be75828282403f61d6e00ac16c73)

File: react/kanban-board/src/contexts/State.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        },
+        lightTheme: {
+            background: 'white',
+            color: 'grey'
+        },
+        currentTheme: {
+            background: 'white',
+            color: 'grey'
+        }
+    }
```
[the remark this answers is not in the document]
Comment (Awsaf780) [reply react/kanban-board/src/contexts/State.js]: Resolved in commit [cbcb445](https://github.com/CraftsmenLtd/coe-batch-q322/pull/290/commits/cbcb445d7dcb9950e8dc6a7b682f92e581983b71)

File: react/kanban-board/src/contexts/State.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        backlog: [
+            {id: Math.random(), task: 'Complete Kanban', status: 'backlog'},
+            {id: Math.random(), task: 'Complete Terraform', status: 'backlog'},
+            {id: Math.random(), task: 'Complete Serverless', status: 'backlog'},
+        ],
+        countBacklog: 0,
+        countProgress: 0,
+        countReview: 0,
+        countReady: 0,
+        countDone: 0,
```
[the remark this answers is not in the document]
Comment (Awsaf780) [reply react/kanban-board/src/contexts/State.js]: Revolved in commit [9331c69](https://github.com/CraftsmenLtd/coe-batch-q322/pull/290/commits/9331c6964014d8ccfbe6ce8f29fa59dc3cd1ae5b)

File: react/kanban-board/src/components/home/components/kanban/section/KanbanSection.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  }
+  else if (props.status === 'inReview') {
+    count = countReview;
+  }
+  else if (props.status === 'ready') {
+    count = countReady;
+  }
+  else if (props.status === 'done') {
+    count = countDone;
+  }
```
[the remark this answers is not in the document]
Comment (Awsaf780) [reply react/kanban-board/src/components/home/components/kanban/section/KanbanSection.js]: Revolved in commit [9331c69](https://github.com/CraftsmenLtd/coe-batch-q322/pull/290/commits/9331c6964014d8ccfbe6ce8f29fa59dc3cd1ae5b)

File: react/kanban-board/src/contexts/State.js [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+            color: 'white'
+        },
+        lightTheme: {
+            background: 'white',
+            color: 'grey'
+        },
+        currentTheme: {
+            background: 'white',
+            color: 'grey'
+        }
```
[the remark this answers is not in the document]
Comment (Awsaf780) [reply react/kanban-board/src/contexts/State.js]: Resolved in commit [4c507bf](https://github.com/CraftsmenLtd/coe-batch-q322/pull/290/commits/4c507bfaee45926710999a29c26e2258e328c331)

File: react/kanban-board/src/components/home/components/kanban/KanbanBoard.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
+import React from 'react'
+import {KanbanBoard} from './Kanban-styled'
+import KanbanSection from './section/KanbanSection'
+
+function Kanban() {
+
+  return (
+    <KanbanBoard>
+        <KanbanSection input={true} type={'Backlog'} status={'backlog'}/>
+        <KanbanSection type={'In Progress'} status={'inProgress'}/>
```
[the remark this answers is not in the document]
Comment (Awsaf780) [reply react/kanban-board/src/components/home/components/kanban/KanbanBoard.js]: [2c97f0b](https://github.com/CraftsmenLtd/coe-batch-q322/pull/290/commits/2c97f0b742c68f88fb6ef0f91a67a8d06f8c7960)

File: react/kanban-board/src/components/home/components/kanban/section/components/taskList/TaskList.js [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+    }
+    else if ((backlog[index].status === 'inReview') && (itemDropDivName[0].name === 'ready')) {
+      UpdateBacklog(id, 'ready');
+    }
+    else if ((backlog[index].status === 'ready') && (itemDropDivName[0].name === 'done')) {
+      UpdateBacklog(id, 'done');
+    }
+
+    updateListCount();
+  }
```
[the remark this answers is not in the document]
Comment (Awsaf780) [reply react/kanban-board/src/components/home/components/kanban/section/components/taskList/TaskList.js]: Resolved in commit [3afe5d6](https://github.com/CraftsmenLtd/coe-batch-q322/pull/290/commits/3afe5d637a7acc805bbd074304528a11128ade0b)

File: react/kanban-board/src/contexts/Reducer.js [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+        ...state,
+        backlog: [...state.backlog, newTask],
+      };
+
+    case RENDER_LATEST:
+      console.log("ReRender Backlog", state.backlog);
+      return {
+        ...state,
+        backlog: [...state.backlog],
+      };
```
Comment (tusher-A) [inline react/kanban-board/src/contexts/Reducer.js:32]: This should be removed.

File: react/kanban-board/src/contexts/Reducer.js [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+        ...state,
+        backlog: state.backlog.filter((todo) => todo.id !== action.payload),
+      };
+
+    case "TOGGLE_THEME":
+      if (state.currentTheme["type"] === "light") {
+        return {
+          ...state,
+          currentTheme: { type: "dark", background: "black", color: "white" },
+        };
```
Comment (tusher-A) [inline react/kanban-board/src/contexts/Reducer.js:45]: Should use proper enum

**Review by tusher-A — CHANGES_REQUESTED** [review]
Some changes are requested.

File: react/kanban-board/src/contexts/Reducer.js (hunk 8 above)
Comment (Awsaf780) [reply react/kanban-board/src/contexts/Reducer.js]: Resolved in commit - [e35d5fc](https://github.com/CraftsmenLtd/coe-batch-q322/pull/290/commits/e35d5fce310e06442df81422bf1722629caee639)

File: react/kanban-board/src/contexts/Reducer.js (hunk 9 above)
Comment (Awsaf780) [reply react/kanban-board/src/contexts/Reducer.js]: Resolved in commit - [e35d5fc](https://github.com/CraftsmenLtd/coe-batch-q322/pull/290/commits/e35d5fce310e06442df81422bf1722629caee639)

**Review by tusher-A — APPROVED** [review]
LGTM

**Pull request merged**